lib/ToolbarBadgeHub.jsm

```diff
@@ -37,6 +42,14 @@ class _ToolbarBadgeHub {
     this.messageRequest("toolbarBadgeUpdate");
   }
 
+  executeAction({ id }) {
+    switch (id) {
+      case "show-whatsnew-button":
+        ToolbarPanelHub.enableToolbarButton();
```

Comment:
I guess we need to implement and call `ToolbarPanelHub.disableToolbarButton()` when the badge message is dismissed 🤔  Right now the toolbar button stays around. Can be a followup.

Reply:
Shouldn't that be the concern of the Panel? We know we want to dismiss it after the very first click. I would instead register a `panel:hidden` event listener in `_showToolbarButton`.

Since the badge disappears when you click the button we would have a situation where an event listener/action is associated with a badge but that badge does not exist/has been dismissed which I find a bit weird.

What do you think?

Reply:
Yeah that's a good point. I'll file a followup to have `ToolbarPanelHub` handle that.